File: app/src/main/java/com/yzyfdf/libsample/ui/HomeFragment.java
package com.yzyfdf.libsample.ui;

import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;

import com.flyco.tablayout.SlidingTabLayout;
import com.yzyfdf.library.base.BaseFragment;
import com.yzyfdf.libsample.R;
import com.yzyfdf.libsample.adapter.TabAdapter;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;

/**
 * @author sjj , 2019/6/5 16:26
 * //todo
 */
public class HomeFragment extends BaseFragment {

    @BindView(R.id.tabLayout)
    SlidingTabLayout mTabLayout;
    @BindView(R.id.viewPager)
    ViewPager mViewPager;

    private boolean isFirst = true;

    public static HomeFragment getInstance() {
        HomeFragment fragment = new HomeFragment();
        return fragment;
    }

    @Override
    protected int getLayoutResource() {
        return R.layout.fragment_home;
    }

    @Override
    public void initPresenter() {

    }

    @Override
    protected void initView() {

    }

    private void getData() {
        isFirst = false;
        ArrayList<Fragment> fragments = new ArrayList<>();
        List<String> titles = new ArrayList<>();

        for (int i = 0; i < 5; i++) {
            fragments.add(InfoFragment.getInstance(true));
            titles.add("标题" + (i + 1));
        }

        TabAdapter tabAdapter = new TabAdapter(getChildFragmentManager(), fragments);
        mViewPager.setAdapter(tabAdapter);
        mViewPager.setOffscreenPageLimit(fragments.size());
        mTabLayout.setViewPager(mViewPager, titles.toArray(new String[titles.size()]));
    }

    @Override
    public void onHiddenChanged(boolean hidden) {
        super.onHiddenChanged(hidden);
        System.out.println("hidden = " + hidden);
        if (isFirst && !hidden) {
            getData();
        }
    }
}

File: app/src/main/java/com/yzyfdf/libsample/ui/HomeFragment.kt
package com.yzyfdf.libsample.ui

import com.flyco.tablayout.listener.OnTabSelectListener
import com.yzyfdf.library.base.BaseFragment
import com.yzyfdf.library.base.BaseModel
import com.yzyfdf.library.base.BasePresenter
import com.yzyfdf.libsample.R
import com.yzyfdf.libsample.adapter.TabAdapter
import kotlinx.android.synthetic.main.fragment_home.*
import java.util.*

/**
 * @author sjj , 2019/6/5 16:26
 * //todo
 */
class HomeFragment(val Index: Int) : BaseFragment<BasePresenter<*, *>, BaseModel>() {


    override val layoutResource: Int
        get() = R.layout.fragment_home

    override fun initPresenter() {

    }

    override fun initView() {
    }

    override fun lazyInit4Fl() {
        getData()
    }

    private fun getData() {
        val fragments = ArrayList<BaseFragment<*, *>>()
        val titles = ArrayList<String>()

        for (i in 0..4) {
            fragments.add(InfoFragment())
            titles.add("标题" + (i + 1))
        }

        val tabAdapter = TabAdapter(childFragmentManager, fragments)
        viewPager.adapter = tabAdapter
        viewPager.offscreenPageLimit = fragments.size
        tabLayout.setViewPager(viewPager, titles.toTypedArray())
        tabLayout.setOnTabSelectListener(object : OnTabSelectListener {
            override fun onTabSelect(position: Int) {

            }

            override fun onTabReselect(position: Int) {
                fragments[position].onPagerReSelected()
            }
        })

    }


}
